### DISCUSSION TITLE
Filtering array element by index

### DISCUSSION THREAD

**Discussion by atfera** [mikro-orm/mikro-orm#2630] [Q&A]
Hi, I'm trying to use mikro-orm in an application (mikro-orm is an amazing project btw).

I just wanted to know if there is an easy way to filter array element by index, ex:

```javascript
@Embeddable()
export class Value {

    @Property()
    id: number;

    @Property({columnType: 'double precision'})
    value: number;
}

@Entity()
export class Data {

    @Embedded(() => Value, { array: true })
    values: Value[] = [];

    @Property({ type: ArrayType, columnType: 'double precision[]' })
    normalArray: number[];
}
```

If I want to filter the first element in `normalArray` or `values`, I need to use `expr()` function like this: (For postgresql)
```javascript
// values 
em.find({
  [expr(`("values"->0->>'value')::float8`)]: { $gt: 0.5 },
  [expr(`("normalArray"[1])`)]: { $gt: 0.5 }
})
```

In Prisma it's possible like [this for json array](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#targeting-an-array-element-by-index), but I didn't go for it because I use a lot of custom types in my application.

Thanks for your answers :)

**Comment by B4nan** [comment]
> In Prisma it's possible like this for json array, but I didn't go for it because I use a lot of custom types in my application.

So their solution is to let users provide the driver dependent part (the example differs for pg and mysql)? So pretty much the same as here, just a different API. 

I guess we could support this out of box at least for some drivers, we already have this abstracted for nested JSON property lookup and embeddables in general.
